core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -255,6 +256,47 @@ class TransactionCoordinator(brokerId: Int,
     }
   }
 
+  def handleDescribeTransactions(
+    transactionalId: String
+  ): DescribeTransactionsResponseData.TransactionState = {
+    val transactionState = new DescribeTransactionsResponseData.TransactionState()
+      .setTransactionalId(transactionalId)
+
+    if (!isActive.get()) {
```

Comment:
Should we check `active` before looping all transaction ids? The other APIs depending on zk controller check 'active' once for whole request.

Reply:
That's a fair point. The difference is that none of the other APIs are batched. I thought it was simpler to let the coordinator API work only with individual transactionalIds and leave the batching aspect to `KafkaApis`. The downside of having this potentially redundant check seems not too bad.